src/modules/navigator/navigator_main.cpp

```diff
@@ -307,25 +316,41 @@ Navigator::run()
 						}
 
 					} else if (PX4_ISFINITE(cmd.param7)) {
+						// Received only a request to change altitude, thus we keep the setpoint
+						rep->current.lat = curr->current.lat;
+						rep->current.lon = curr->current.lon;
+						rep->current.alt = cmd.param7;
+
+					} else {
+						// All three set to NaN - pause vehicle
+						rep->current.alt = get_global_position()->alt;
 
-						// Altitude without position change
-						// This condition is necessary for altitude changes just after takeoff where lat and lon are still nan
-						if (curr->current.valid && PX4_ISFINITE(curr->current.lat) && PX4_ISFINITE(curr->current.lon)) {
-							rep->current.lat = curr->current.lat;
-							rep->current.lon = curr->current.lon;
+						if (_vstatus.vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING
+						    && (get_position_setpoint_triplet()->current.type != position_setpoint_s::SETPOINT_TYPE_TAKEOFF)) {
+
+							// For multirotors we need to account for the braking distance, otherwise the vehicle will overshoot and go back
+							double lat, lon;
+							float course_over_ground = atan2f(_local_pos.vy, _local_pos.vx);
+
+							// predict braking distance
+
+							const float velocity_hor_abs = sqrtf(_local_pos.vx * _local_pos.vx + _local_pos.vy * _local_pos.vy);
+
+							float multirotor_braking_distance = math::trajectory::computeBrakingDistanceFromVelocity(velocity_hor_abs,
+											    _param_mpc_jerk_max, _param_mpc_acc_hor, 0.6f * _param_mpc_jerk_max);
+
+							waypoint_from_heading_and_distance(get_global_position()->lat, get_global_position()->lon, course_over_ground,
+											   multirotor_braking_distance, &lat, &lon);
+							rep->current.lat = lat;
+							rep->current.lon = lon;
+							rep->current.yaw = get_local_position()->heading;
```

Comment:
@cmic0 it seems that if you want to just move in the yaw axis while in place you can't do it due to this line

Reply:
@garfieldG in hold mode you can't control vehicle orientation/attitude, by default we have RC_OVERRIDE enabled which would switch you to Position in case you move the yaw with a stick, or maybe i am not getting your use case :thinking: ?